src/openassetio-core/log/ConsoleLogger.cpp

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#include <iomanip>
+#include <iostream>
+
+#include <openassetio/log/ConsoleLogger.hpp>
+
+// Foreground ANSI color codes (combined with \033[<color>m)
+constexpr const int kSeverityColors[] = {36, 32, 32, 0, 33, 31, 31};
+
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+namespace log {
+ConsoleLoggerPtr ConsoleLogger::make(bool colorOutput) {
+  return std::shared_ptr<ConsoleLogger>(new ConsoleLogger(colorOutput));
+}
+
+ConsoleLogger::ConsoleLogger(bool colorOutput) : colorOutput_(colorOutput) {}
+
+void ConsoleLogger::log(Severity severity, const Str& message) {
+  if (colorOutput_) {
+    std::cerr << "\033[0;" << kSeverityColors[severity] << "m";
+  }
+
+  // NOLINTNEXTLINE(readability-magic-numbers)
+  std::cerr << std::setw(11) << kSeverityNames[severity] << ": " << message;
+
+  if (colorOutput_) {
+    std::cerr << "\033[0m";
+  }
+
+  std::cerr << std::endl;
```

Comment:
[fix] `std::endl` does a `\n` followed by a `std::flush`.  As mentioned in the other comment, this is unnecessary, especially for `cerr`, and has performance implications.  

Also frowned upon by [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rio-endl):
>  it simply slows down output by doing redundant flush()s.
```suggestion
  std::cerr << "\n";
```

Reply:
Updated [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/552/commits/40e5b149e5a972fdb0596987c1a80e23f680c125).